### ISSUE TITLE
Execution stopped without error message

### ISSUE THREAD

**Issue by timoth500** [project8/dragonfly#115]
[CRITICAL] dragonfly.subcommands.run_scripting(331) ->

This line displays at the end of a script in dry run mode only when -f flag is used under certain conditions

**Comment by laroque** [comment]
Is there some context? Non-error logger output from prior to that line? Did it really not print anything after the `->`? Where there any issues with any of the running services from around the same time?

**Comment by wcpettus** [comment]
I never saw any slack alert associated with this.  Does this mean the amqp handler or slack interface suppress empty logger messages?

**Comment by guiguem** [comment]
@wcpettus yes it seems that no slack message is sent: I would need to look how slack interface interprets empty payload field.
Another question is why the run scripting log empty messages.

This problem happened quite randomly: once when setting the current of a trap coil, once when setting the frequency mask on the rsa.
No error before this message.
Executing the script right where it stopped allowed to overcome the point where it stopped.
Not sure what is going on...

**Comment by wcpettus** [comment]
when the slack api is given an empty msg.payload, it won't post:
```
2017-10-01T16:05:28[DEBUG   ] dragonfly.implementations.slack_interface(102) -> api call returned:{u'ok': False, u'error': u'no_text'}
```
Commit afeda623ccb9ab5aeed8d1e4aafcf73423fb1080 bypasses this by overwriting the payload with `None`

**Comment by wcpettus** [comment]
The slack notification is fixed.
The crash is unknown, but there isn't enough information to work with, so this issue is closed for now with `wontfix`

**Comment by guiguem** [comment]
This error happened again today: ```
(drag_v1.9.3_drip_wp2.1.1_python3.5.8_hard_v1.11.3_fire_v1.1.1_swarm_v1.4.9)project8@claude:~/Repos/grasshopper/Execution_scripts/esr_commissioning$ dragonfly execute esr_cell_temp_dependence.yaml -b myrna.p8 -vvv
warning: slack is only ever warning, setting that
2017-11-18T16:49:18[DEBUG   ] dragonfly(268) -> calling <dragonfly.subcommands.run_scripting.RunScript object at 0x7f3b075c5b50> with args:
{'verbose': 3, 'dry_run': False, 'force_restart': False, 'execution_file': 'esr_cell_temp_dependence.yaml', 'broker': 'myrna.p8', 'tmux': None, 'func': <dragonfly.subcommands.run_scripting.RunScript object at 0x7f3b075c5b50>, 'config': None}
2017-11-18T16:49:18[INFO    ] dragonfly.subcommands.run_scripting(249) -> new cache file is /tmp/esr_cell_temp_dependence_cache.json
2017-11-18T16:49:18[INFO    ] dragonfly.subcommands.run_scripting(313) -> doing action [0]: set
2017-11-18T16:49:18[INFO    ] dragonfly.subcommands.run_scripting(398) -> doing set block
2017-11-18T16:49:18[INFO    ] dragonfly.subcommands.run_scripting(405) -> setting all_trap_coils->1.0
2017-11-18T16:49:18[DEBUG   ] dripline.core.service(524) -> request to send to <all_trap_coils> is: {
    "timestamp": "2017-11-19T00:49:18Z", 
    "sender_info": {
        "username": "project8", 
        "exe": "/home/project8/python_environments/drag_v1.9.3_drip_wp2.1.1_python3.5.8_hard_v1.11.3_fire_v1.1.1_swarm_v1.4.9/bin/dragonfly", 
        "package": "dripline", 
        "service_name": "execution_script", 
        "hostname": "claude", 
        "version": "wp2.1.1", 
        "commit": "g291e931"
    }, 
    "payload": {
        "values": [
            1.0
        ]
    }, 
    "msgop": 0
}
2017-11-18T16:49:18[CRITICAL] dragonfly.subcommands.run_scripting(331) -> 
2017-11-18T16:49:18[INFO    ] dripline.core.service(548) -> sending an alert
2017-11-18T16:49:18[DEBUG   ] dripline.core.service(552) -> to status_message.critical.dripline sending {
    "timestamp": "2017-11-19T00:49:18Z", 
    "payload": "", 
    "sender_info": {
        "username": "project8", 
        "exe": "/home/project8/python_environments/drag_v1.9.3_drip_wp2.1.1_python3.5.8_hard_v1.11.3_fire_v1.1.1_swarm_v1.4.9/bin/dragonfly", 
        "package": "dripline", 
        "service_name": "unknown_service_65333fb-66d", 
        "hostname": "claude", 
        "version": "wp2.1.1", 
        "commit": "g291e931"
    }
}
(drag_v1.9.3_drip_wp2.1.1_python3.5.8_hard_v1.11.3_fire_v1.1.1_swarm_v1.4.9)project8@claude:~/Repos/grasshopper/Execution_scripts/esr_commissioning$ ```
All the services were restarted a couple of hours ago, some get, set and cmd were done a couple of minutes earlier

**Comment by guiguem** [comment]
The execution script used the `all_trap_coils` (which is not a correct endpoint) into `all_trap_coil`(which is a correct one). And the error vanished...
If this is really the cause of the issue, it might come from the multi_do that does not catch the error properly

**Comment by wcpettus** [comment]
The empty return_msg was an issue in dripline-python (fixed in 3.5.9), and the error messages have also been improved in dragonfly v1.9.4
